diff --git a/package/sveltekit-search-params.svelte.js b/package/sveltekit-search-params.svelte.js
index e039b465c422f009a0a83f1d8ae41fed1856fcdd..8b41ffb6ce19adad5599da27fb36aec9c0253689 100644
--- a/package/sveltekit-search-params.svelte.js
+++ b/package/sveltekit-search-params.svelte.js
@@ -1,6 +1,18 @@
-import { browser, building } from '$app/environment';
+import { VERSION } from '@sveltejs/kit';
 import { goto } from '$app/navigation';
 import { page } from '$app/state';
+
+/**
+ * SvelteKit 3 changes the goto options.
+ * NOTE: Simplify the GOTO_OPTIONS once this library no longer supports SvelteKit 2.
+ */
+const is_sveltekit_2 = VERSION.startsWith('2.');
+
+// NOTE: Simplify this once this library no longer supports SvelteKit 2.
+const { browser, building } = await (is_sveltekit_2
+	? import('$app/environment')
+	: import('$app/env'));
+
 // during building we fake the page url with no search params as it should be
 // during prerendering. This allow the application to still build and the client
 // side behavior is still persisted after the build
@@ -15,16 +27,23 @@ function is_complex_equal(current, next, equality_fn = (current, next) => JSON.s
         typeof next === 'object' &&
         equality_fn(current, next));
 }
-const GOTO_OPTIONS = {
-    keepFocus: true,
-    noScroll: true,
-    replaceState: true,
-};
-const GOTO_OPTIONS_PUSH = {
-    keepFocus: true,
-    noScroll: true,
-    replaceState: false,
-};
+const GOTO_OPTIONS = is_sveltekit_2
+	? {
+			keepFocus: true,
+			noScroll: true,
+			replaceState: true,
+		}
+	: {
+			reset: false,
+			replace: true,
+		};
+
+const GOTO_OPTIONS_PUSH = is_sveltekit_2
+	? { keepFocus: true, noScroll: true, replaceState: false }
+	: {
+			reset: false,
+			replace: false,
+		};
 const batched_updates = new Set();
 let batch_timeout;
 const debounced_timeouts = new Map();
@@ -99,7 +118,7 @@ function create_root_navigator(root, encodes, overrides, navigation_options) {
             current = current[piece];
         }
         current[key] = to_set;
-        do_navigate(name, 
+        do_navigate(name,
         // this allow for `push` to work
         Array.isArray(value) && key === 'length'
             ? structuredClone(value)
diff --git a/package.json b/package.json
index 515bc60772a8252778f894faf6bdd99e65108452..46f810695ef9e55c1cdf251558672d961208585a 100644
--- a/package.json
+++ b/package.json
@@ -49,7 +49,7 @@
   },
   "type": "module",
   "peerDependencies": {
-    "@sveltejs/kit": "^2.12.0",
+    "@sveltejs/kit": "^2.12.0 || >=3.0.0-next.24 || ^3.0.0",
     "svelte": "^5.0.0"
   },
   "exports": {
